packages/search-input/src/SearchInput/SearchInput.types.ts

```diff
@@ -62,6 +62,33 @@ interface BaseSearchInputProps
    * The placeholder text shown in the input field before the user begins typing.
    */
   placeholder?: string;
+
+  /**
+   * For internal uses, expose a slot to render a custom div to wrap the input
+   * @internal
+   */
+  __INTERNAL__divWrapperSlot__?: React.ForwardRefExoticComponent<
```

Comment:
I disagree with this approach - I think there should be one slot prop that replaces the entire container.

Reply:
This pattern is maintaining parity with Select. The difference is that unlike Select where only the Button needs to be overridden, this component requires props on both the div and input to be overridden.

Reply:
WE only have ___INTERNAL___MenuButtonSlot - i think we should maintain the same idea here.

Reply:
We should have one line of code that says internal ? <Internal {...receivedProps} /> : <SearchInput />

Reply:
I don't think that would work since a lot of the functionality is dependent on both `searchBoxRef` and `inputRef`, unless I'm missing something

Reply:
We need to be able to pass in both a custom searchBox and a custom input. If we completely replace the existing searchBox, the input will be completely replaced as well. The other option is that if we replace it completely, we'd be passing in a `searchBoxRef` AND an `inputRef`. I think using two slot props would be better than passing two refs to the override component.

Reply:
you can't have `<Internal searchBoxRef={forwardToSearchBox} inputRef={forwardToInput} />` ?

Reply:
@shaneeza what do you think?

Reply:
is it easy to forward multiple refs?

Reply:
I don't believe it's easy. It would be kind of hacky from what I've seen

Reply:
ok will defer to you two